src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -180,22 +199,14 @@ private void traceRequest(final RequestLogInfo requestLogInfo, final int statusC
         } else if (requestLogInfo.contentLength < 5000) {
             sloBucket = "<5K";
         }
-
-        GlobalTracer.get()
-                .buildSpan("publish_events")
-                .withStartTimestamp(TimeUnit.MILLISECONDS.toMicros(requestLogInfo.requestTime))
-                .start()
-                .setTag("client_id", requestLogInfo.user)
-                .setTag("event_type", eventType)
-                .setTag("error", statusCode == 207 || statusCode >= 500)
-                .setTag("http.status_code", statusCode)
-                .setTag("http.url", requestLogInfo.path + requestLogInfo.query)
-                .setTag("http.header.content_encoding", requestLogInfo.contentEncoding)
-                .setTag("http.header.accept_encoding", requestLogInfo.acceptEncoding)
-                .setTag("http.header.user_agent", requestLogInfo.userAgent)
-                .setTag("slo_bucket", sloBucket)
-                .setTag("content_length", requestLogInfo.contentLength)
-                .finish(TimeUnit.MILLISECONDS.toMicros(requestLogInfo.requestTime + timeSpentMs));
+        final String eventType = requestLogInfo.path.substring("/event-types/".length(),
+                requestLogInfo.path.lastIndexOf("/events"));
+        TracingService.setCustomTags(scope, ImmutableMap.<String, Object>builder()
+                .put("event_type", eventType)
+                .put("http.status_code", statusCode)
+                .put("slo_bucket", sloBucket)
+                .put("content_length", requestLogInfo.contentLength)
+                .put("error", statusCode == 207 || statusCode >= 500).build());
```

Comment:
Isn't `error` also set in the controller?

Reply:
Then I guess I missed something. Why is there a "filter" span and a "controller" span? Can't we have a single "parent" span that lasts throughout the entire publishing operation?

Reply:
I see your point. I was hoping to have a child span starting at the controller level but since it does not add much value alone we can combine it. Fixed in bceb690